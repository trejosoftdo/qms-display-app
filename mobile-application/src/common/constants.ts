export const AUTHORIZATION_PENDING_CODE = 'authorization_pending';

export const VALIDATION_ERROR_STATUS_CODE = 400;

export const DEVICE_NOT_CONNECTED_ERROR_MESSAGE = 'Device is not connected';

export const CONNECTION_EXPIRED_ERROR_MESSAGE = 'Connection expired!';

export const APP_PROTO = 'app://';

export const DEVICE_ID_KEY = 'SECURE_DEVICE_ID';

export const CONNECTION_DETAILS_KEY = 'CONNECTION_DETAILS';

export const WEB_PLATFORM = 'web';

export const BEARER_PORTION = 'Bearer';

export const EMPTY_VALUE = '';

export const DEFAULT_REGULAR_FONT_NAME = 'Montserrat-Regular';

export const DEFAULT_LIGHT_FONT_NAME = 'Montserrat-Light';

export const DEFAULT_MEDIUM_FONT_NAME = 'Montserrat-Medium';

export const NORMAL_FONT_WEIGHT = 'normal';

export const ONE_SECOND_MILISECONDS = 1000;

export const INTERVAL_TIME = 10 * ONE_SECOND_MILISECONDS;

export const SERVICES_PATH = '/services';

export const TICKET_DETAILS_PATH = '/ticket-details';

export const HOME_PATH = '/';

export const CATEGORIES_PATH = '/categories';

export const CONFIGURATION_PATH = '/configuration';

export const NO_OP = () => {};


export const LANGUAGES = {
  SPANISH: 'es',
  ENGLISH: 'en',
};

export const CONFIG_OPTIONS = {
  INFORMATION: 'information',
  CONNECTION: 'connection',
  LANGUAGE: 'language',
};
