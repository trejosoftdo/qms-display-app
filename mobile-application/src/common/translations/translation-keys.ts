export const TRANSLATION_CATEGORIES_KEY = 'translation:categories';

export const TRANSLATION_CONFIGURATION_KEY = 'translation:configuration';

export const TRANSLATION_CONNECTION_KEY = 'translation:connection';

export const TRANSLATION_INFORMATION_KEY = 'translation:information';

export const TRANSLATION_SERVICES_KEY = 'translation:services';

export const TRANSLATION_TICKET_DETAILS_KEY = 'translation:turnDetails';

export const TRANSLATION_LANGUAGE_KEY = 'translation:language';

export const TRANSLATION_CUSTOMER_DATA_KEY = 'translation:customerData';

export const TRANSLATION_WAIT_MESSAGE_KEY = 'translation:waitMessage';

export const TRANSLATION_CHOOSE_CATEGORY_MESSAGE_KEY = 'translation:chooseCategoryMessage';

export const TRANSLATION_CHOOSE_SERVICE_MESSAGE_KEY = 'translation:chooseServiceMessage';

export const TRANSLATION_DETAILS_KEY = 'translation:details';

export const TRANSLATION_ID_KEY = 'translation:id';

export const TRANSLATION_SPANISH_KEY = 'translation:spanish';

export const TRANSLATION_ENGLISH_KEY = 'translation:english';

export const TRANSLATION_LANGUAGE_MESSAGE_KEY = 'translation:languageMessage';

export const TRANSLATION_CUSTOMER_DATA_FORM_MESSAGE_KEY = 'translation:customerDataFormMessage';

export const TRANSLATION_NAME_KEY = 'translation:name';

export const TRANSLATION_CONTINUE_KEY = 'translation:continue';

export const TRANSLATION_WELCOME_TURN_KEY = 'translation:welcomeTurn';

export const TRANSLATION_USERS_IN_QUEUE_KEY = 'translation:usersInQueue';

export const TRANSLATION_PRINT_TICKET_KEY = 'translation:printTicket';

export const TRANSLATION_OPTIONS_KEY = 'translation:options';

export const TRANSLATION_CONFIG_MESSAGE_KEY = 'translation:configMessage';

export const TRANSLATION_RECONNECT_MESSAGE_KEY = 'translation:reconnect';

export const TRANSLATION_RETURN_MESSAGE_KEY = 'translation:return';

export const TRANSLATION_CONNECT_KEY = 'translation:connect';

export const TRANSLATION_APPLICATION_KEY = 'translation:application';

export const TRANSLATION_CONNECTED_MESSAGE_KEY = 'translation:alreadyConnectedMessage';

export const TRANSLATION_SUCCESS_CONNECTION_MESSAGE_KEY = 'translation:successConnectionMessage';

export const TRANSLATION_TO_CONNECT_MESSAGE_KEY = 'translation:toConnectMessage';

export const TRANSLATION_INITIAL_CONNECT_MESSAGE_KEY = 'translation:initialConnetMessage';