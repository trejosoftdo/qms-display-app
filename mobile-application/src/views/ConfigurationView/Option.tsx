import * as React from 'react';
import { List } from 'react-native-paper';

/**
 * OptionProps defines the props for the Option Component.
 *
 * @property title - option title
 * @property icon - option icon
 * @property onPress - option on press handler
 */
interface OptionProps {
  title: string;
  icon: string;
  onPress: () => void;
}

/**
 * A component a configuration option
 *
 * @param {OptionProps} props - The props for the Option component.
 */
const Option: React.FC<OptionProps> = (props: OptionProps) => (
  <List.Item
    title={props.title}
    left={() => <List.Icon icon={props.icon} />}
    right={() => <List.Icon icon="chevron-right" />}
    onPress={props.onPress}
  />
);

export default Option;