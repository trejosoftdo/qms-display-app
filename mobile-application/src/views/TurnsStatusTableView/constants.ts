export const BEING_ATTENDED_STATUS_CODE = "BEING_ATTENDED";

export const TO_BE_ATTENDED_STATUS_CODE = "TO_BE_ATTENDED";