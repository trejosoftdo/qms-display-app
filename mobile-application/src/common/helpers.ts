import { Platform } from 'react-native';
import { router } from 'expo-router';
import { APP_PROTO, EMPTY_VALUE, ONE_SECOND_MILISECONDS, WEB_PLATFORM } from './constants';

/** 
 * Checks if the platform is web
 * @returns boolean
 */
export const isWeb = (): boolean => Platform.OS === WEB_PLATFORM;

/** 
 * Gets the current time
 * @returns number
 */
export const getCurrentTime = (): number => new Date().getTime();

/**
 * Calculates the expire time
 * 
 * @param  {number} expiresIn
 * @returns number
 */
export const calculateExpireTime = (expiresIn: number): number => getCurrentTime() + expiresIn * ONE_SECOND_MILISECONDS;


/** 
 * Extracs icon name
 * @returns string
 */
export const extractIconName = (iconUrl: string): string => iconUrl.replace(APP_PROTO, EMPTY_VALUE)

/**
 * Navigates to a given path
 * 
 * @param  {string} pathname
 * @param  {any} params
 * @returns void
 */
export const goToPath = (pathname: string, params: Record<string, any> = undefined): void => {
  router.push({
    pathname,
    params,
  });
};

/**
 * Waits for certain time
 * @param  {number} timeout 
 * @returns Promise<any>
 */
export const delay = (timeout: number) => new Promise((resolve) => {
  setTimeout(resolve, timeout);
});

/**
 * Gets the response blob url
 * @param  {Response} response 
 * @returns Promise<string>
 */
export const getResponseBlobUrl = async (response: Response): Promise<string> => {
  const stream = await response.blob();
  return URL.createObjectURL(stream);
};
