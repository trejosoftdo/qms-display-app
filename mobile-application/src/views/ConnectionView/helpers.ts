import { ConnectionData, ConnectionDetails } from "../../common/models";
import { TRANSLATION_CONNECTED_MESSAGE_KEY, TRANSLATION_INITIAL_CONNECT_MESSAGE_KEY, TRANSLATION_SUCCESS_CONNECTION_MESSAGE_KEY, TRANSLATION_TO_CONNECT_MESSAGE_KEY } from "../../common/translations/translation-keys";

/**
 * Gets the correct message key from the data state
 * 
 * @param  {ConnectionData} data
 * @returns string
 */
export const getMessageKey = (data: ConnectionData, connectionDetails?: ConnectionDetails): string => {
  if (connectionDetails?.accessToken) {
    return TRANSLATION_CONNECTED_MESSAGE_KEY;
  }

  if (data?.tokens?.accessToken) {
    return TRANSLATION_SUCCESS_CONNECTION_MESSAGE_KEY;
  }

  return data?.userCode ? TRANSLATION_TO_CONNECT_MESSAGE_KEY : TRANSLATION_INITIAL_CONNECT_MESSAGE_KEY;
};
