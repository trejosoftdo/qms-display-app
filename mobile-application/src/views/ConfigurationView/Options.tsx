
import * as React from 'react';
import { StyleSheet } from 'react-native';
import { List } from 'react-native-paper';
import { useTranslation } from 'react-i18next';
import { CONFIG_OPTIONS } from '../../common/constants';
import { TRANSLATION_CONNECTION_KEY, TRANSLATION_INFORMATION_KEY, TRANSLATION_LANGUAGE_KEY, TRANSLATION_OPTIONS_KEY } from '../../common/translations/translation-keys';
import Option from './Option';

/**
 * OptionsProps defines the props for the Options Component.
 *
 * @property onOptionSelect - option selection handler
 */
interface OptionsProps {
  onOptionSelect: (name: string) => void;
};

/**
 * A component a configuration options
 *
 * @param {OptionsProps} props - The props for the Options component.
 */
const Options: React.FC<OptionsProps> = (props: OptionsProps) => {
  const { t } = useTranslation();
  return (
    <List.Section style={styles.container}>
      <List.Subheader>{t(TRANSLATION_OPTIONS_KEY)}</List.Subheader>
      <Option
        title={t(TRANSLATION_INFORMATION_KEY)}
        icon="information"
        onPress={() => props.onOptionSelect(CONFIG_OPTIONS.INFORMATION)}
      />
      <Option
        title={t(TRANSLATION_CONNECTION_KEY)}
        icon="connection"
        onPress={() => props.onOptionSelect(CONFIG_OPTIONS.CONNECTION)}
      />
      <Option
        title={t(TRANSLATION_LANGUAGE_KEY)}
        icon="translate"
        onPress={() => props.onOptionSelect(CONFIG_OPTIONS.LANGUAGE)}
      />
    </List.Section>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    maxWidth: 500,
  },
});

export default Options;