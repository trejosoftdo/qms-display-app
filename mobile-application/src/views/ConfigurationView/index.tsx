import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { TRANSLATION_CONFIG_MESSAGE_KEY } from '../../common/translations/translation-keys';
import { goToPath } from '../../common/helpers';
import { AppView } from '../../common/components';
import Options from './Options';


/**
 * ConfigurationViewProps defines the props for the Configuration View Component.
 *
 */
interface ConfigurationViewProps {}


/**
 * A component for the configuration view
 *
 * @param {ConfigurationViewProps} props - The props for the Configuration View component.
 */
const ConfigurationView: React.FC<ConfigurationViewProps> = (props: ConfigurationViewProps) => {
  const { t } = useTranslation();
  return (
    <AppView loading={false} headerMessage={t(TRANSLATION_CONFIG_MESSAGE_KEY)} >
      <Options
        onOptionSelect={(option) => {
          goToPath(`/${option}`);
        }}
      />
    </AppView>
  );
};

export default ConfigurationView;