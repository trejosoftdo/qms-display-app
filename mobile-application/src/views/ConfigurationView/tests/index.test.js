import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { goToPath } from '../../../common/helpers';
import ConfigurationView from '../';
import { TRANSLATION_CONNECTION_KEY, TRANSLATION_INFORMATION_KEY, TRANSLATION_LANGUAGE_KEY } from '../../../common/translations/translation-keys';
import { CONFIG_OPTIONS } from '../../../common/constants';


jest.mock('../../../common/helpers');


describe('ConfigurationView component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders the component as expected', () => {
    expect(renderShallow(<ConfigurationView />)).toMatchSnapshot();
    expect(goToPath).toHaveBeenCalledTimes(0);
  });

  it('navigates to information path when selecting that option', () => {
    const { getByText } = render(<ConfigurationView />);
    const element = getByText(`Translated[${TRANSLATION_INFORMATION_KEY}]({})`);

    fireEvent.press(element);

    expect(goToPath).toHaveBeenCalledTimes(1);
    expect(goToPath).toHaveBeenCalledWith(
      `/${CONFIG_OPTIONS.INFORMATION}`
    );
  });

  it('navigates to connection path when selecting that option', () => {
    const { getByText } = render(<ConfigurationView />);
    const element = getByText(`Translated[${TRANSLATION_CONNECTION_KEY}]({})`);

    fireEvent.press(element);

    expect(goToPath).toHaveBeenCalledTimes(1);
    expect(goToPath).toHaveBeenCalledWith(
      `/${CONFIG_OPTIONS.CONNECTION}`
    );
  });

  it('navigates to language path when selecting that option', () => {
    const { getByText } = render(<ConfigurationView />);
    const element = getByText(`Translated[${TRANSLATION_LANGUAGE_KEY}]({})`);

    fireEvent.press(element);

    expect(goToPath).toHaveBeenCalledTimes(1);
    expect(goToPath).toHaveBeenCalledWith(
      `/${CONFIG_OPTIONS.LANGUAGE}`
    );
  });
});
