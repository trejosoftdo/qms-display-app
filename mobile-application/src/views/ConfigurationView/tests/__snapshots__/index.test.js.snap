// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationView component renders the component as expected 1`] = `
<AppView
  headerMessage="Translated[translation:configMessage]({})"
  loading={false}
>
  <Options
    onOptionSelect={[Function]}
  />
</AppView>
`;
